content/docs/rfcs/32/README.md

```diff
@@ -0,0 +1,565 @@
+Spamming by certain entities can worsen the user experience for the other entities in the application, 
+and in some cases can be detrimental for the applications.
+
+RLN (Rate limiting nullfier) is a construct based on zero-knowledge proofs that enables spam prevention mechanism for decentralized, anonymous environments. 
+In anonymous environments, the identity of the entities is unknown, and thus it is very hard to prevent spam attacks. 
+Many applications would benefit from anonymity, but the potential for spam could worsen their UX to a large extent. 
+RLN allows for anonymous signaling while enabling spam prevention on protocol level.
+
+
+# Technical overview
+
+The main RLN construct is implemented using a [ZK-SNARK](https://z.cash/technology/zksnarks/) circuit. 
+However it is helpful to describe the other necessary outside components for interaction with the circuit, 
+which together with the ZK-SNARK circuit enable the above mentioned features.
+
+
+## Terminology
+
+| Term                      | Description                                                                         |
+|---------------------------|-------------------------------------------------------------------------------------|
+| **ZK-SNARK**    | https://z.cash/technology/zksnarks/              |
+| **Stake**                 | Financial or social stake required for registering in the RLN applications. The staking requirement is necessary for disincentivising the end users to spam. A common stake examples are: locking cryptocurrency (financial), linking reputable social identity. |
+| **RLN Registration**    | The process of the user registering themselves to the RLN protocol. The registrations can occur onchain or offchain. Financial or social stake is required for registering to the protocol.   |
+| **Slashing**              | The process of retreiving the secret hash from the user that spams, and from the secret hash generating the identity commitment. Having the secret hash of a user implies that anyone could reveal their "identity" and also retreive their financial stake if it is enabled by the app.      |
+| **Identity nullifier**    | Random 32 byte value used as component for identity secret generation.              |
+| **Identity trapdoor**     | Random 32 byte value used as component for identity secret generation.              |
+| **Identity secret**       | An array of two unique random components (identity nullifier and identity trapdoor), which must be kept secret from the users. Secret hash and identity commitment are derived from this array.        |
+| **Identity secret hash**  | The hash of the identity secret, obtained using the Poseidon hash function. It is used for deriving the identity commitment of the user, and as a private input for zk proof generation. The secret hash should be treated as "private key" for the RLN protocol. The anonymity and security of the user is dependent on the secret hash.                                       |
```

Comment:
Can you please on this part? the "private key" used in the first sentence implies some level of secrecy for the secret hash but the next sentence 
> "The anonymity and security of the user is dependent ..." 

defeats this implication. Wondering should the secret hash be kept private or not?

Reply:
it should be definitely kept private, I should formulate this better as this sentence is confusing